### DISCUSSION TITLE
Add SAMPLE_SIZE to read_json_objects or json_group_structure

### DISCUSSION THREAD

**Discussion by criccomini** [duckdb/duckdb#6018] [Ideas]
I would like to infer schema from a large newline-delimited JSON file. For performance reasons, I would prefer to sample only a small subset of the records to infer the schema. `read_csv_auto` already supports SAMPLE_SIZE. I would like this same behavior, but for JSON. Whether SAMPLE_SIZE is added to `read_json_objects` or `json_group_structure`, I leave to you.

**Comment by lnkuiper** [comment]
I am currently implementing `read_json_auto` in this PR: #5992. This will be used in the same way as `read_csv_auto` and will also have a `SAMPLE_SIZE` parameter, and should be in the next DuckDB release.

For now, you could do:
```sql
SELECT json_group_structure(json)
FROM (SELECT json FROM read_json_objects('<my_file>') LIMIT <my_sample_size>);
```
